Open the AI settings in a side panel in Notebook application

brichet opened the pull request:
Fixes #1250 

Opens the AI settings panel as a side panel, in the left area, when in Notebook application.

This PR adds a dependency to `@jupyter-notebook/application` to know if the shell is INotebookShell. In this case, the settings panel is not opened as a main area widget but in the side panel.

https://github.com/user-attachments/assets/2520b1d3-10af-4675-aef1-dbd525e20f51

In the long run, the settings could be including in the jupyterlab settings, as discussed in #1250. If so, this change may be removed, along with the extra dependency.

dlqqq reviewed (approved):
@brichet Thank you for building this support! Code changes look minimal & safe for non-Notebook users, so I'm comfortable merging this as long as you have tested this PR.

BTW, does this PR need to be backported to 2.x? I don't think so, but want to confirm.

brichet commented:
I tested it successfully with Notebook 7.3.3.

> BTW, does this PR need to be backported to 2.x? I don't think so, but want to confirm.

If it doesn't change since the last time I used the version `2.x`, the settings are included in the chat itself, so it should be backported.

dlqqq commented:
@brichet 

> so it should be backported.

Wait, you mean it should **not** be backported, right? The AI settings are accessed through the chat itself in JAI v2, so JAI v2's settings should already work in Notebook 7.

If you're able to verify that a user can open the AI settings in JAI v2 + Notebook 7, then you can merge this PR and close this thread. 👍
